third_party/nvfuser/python/__init__.py

```diff
@@ -10,14 +10,24 @@ def __exit__(self, type, value, traceback):
     def definition(self):
         raise NotImplementedError("definition() should be implemented by child class!")
 
-    def execute(self, inputs):
-        # if definition is not previously defined by a context manager try a child class
+    def schedule(self):
+        raise NotImplementedError("schedule() should be implemented by child class!")
+
+    def execute(self, inputs, override_user_schedule=False):
+        # if definition is not defined by a context manager, try a child class
         if self.id() is None:
             self._setup_definition()
             self.definition()
             self._finalize_definition()
+            self.func_based_def = True
```

Comment:
This looks a little bit ugly. I think we only `define` a fusion inside `setup_definition` and `schedule` a tensor within `setup_schedule`.
Maybe we can setup flags in those instead of rely on a user set flag...

The assert in `python_binding.cpp` also feels awkward.

Reply:
This was changed.  `func_based_def` is no longer part of an exposure from C++.  It is only book keeping in `execute()`.